packages/react-google-maps-api/src/GoogleMap.tsx

```diff
@@ -28,50 +28,91 @@ const eventMap = {
 }
 
 const updaterMap = {
-  extraMapTypes (map, extra) {
-    extra.forEach(it => {
-      map.mapTypes.set(...it)
+  extraMapTypes (map: google.maps.Map, extra: google.maps.MapType[]) {
+    extra.forEach((it, i) => {
+      map.mapTypes.set(String(i), it)
     })
   },
-  center (map, center) {
+  center (map: google.maps.Map, center: LatLng) {
     map.setCenter(center)
   },
-  clickableIcons (map, icons) {
-    map.setClickableIcons(...icons)
+  clickableIcons (map: google.maps.Map, clickable: boolean) {
+    map.setClickableIcons(clickable)
   },
-  heading (map, heading) {
+  heading (map: google.maps.Map, heading: number) {
     map.setHeading(heading)
   },
-  mapTypeId (map, mapTypeId) {
+  mapTypeId (map: google.maps.Map, mapTypeId: string) {
     map.setMapTypeId(mapTypeId)
   },
-  options (map, options) {
+  options (map: google.maps.Map, options: google.maps.MapOptions) {
     map.setOptions(options)
   },
-  streetView (map, streetView) {
+  streetView (map: google.maps.Map, streetView: google.maps.StreetViewPanorama) {
     map.setStreetView(streetView)
   },
-  tilt (map, tilt) {
+  tilt (map: google.maps.Map, tilt: number) {
     map.setTilt(tilt)
   },
-  zoom (map, zoom) {
+  zoom (map: google.maps.Map, zoom: number) {
     map.setZoom(zoom)
   }
 }
 
-export class GoogleMap extends PureComponent {
-  static propTypes = GoogleMapPropTypes
-  static defaultProps = {
+interface GoogleMapState {
+  map: google.maps.Map
+}
+
+interface GoogleMapProps {
+  id: string;
+  reuseSameInstance?: boolean;
+  mapContainerStyle?: React.CSSProperties;
+  mapContainerClassName?: string;
+  options?: google.maps.MapOptions;
+  extraMapTypes?: google.maps.MapType[];
+  center?: LatLng;
+  clickableIcons?: boolean;
+  heading?: number;
+  mapTypeId?: string;
+  streetView?: google.maps.StreetViewPanorama;
+  tilt?: number;
+  zoom?: number;
+  onClick?: (e: MouseEvent) => void;
+  onDblClick?: (e: MouseEvent) => void;
+  onDrag?: () => void;
+  onDragEnd?: () => void;
+  onDragStart?: () => void;
+  onMapTypeIdChanged?: () => void;
+  onMouseMove?: (e: MouseEvent) => void;
+  onMouseOut?: (e: MouseEvent) => void;
+  onMouseOver?: (e: MouseEvent) => void;
+  onRightClick?: (e: MouseEvent) => void;
+  onTilesLoaded?: () => void;
+  onBoundsChanged?: () => void;
+  onCenterChanged?: () => void;
+  onHeadingChanged?: () => void;
+  onIdle?: () => void;
+  onProjectionChanged?: () => void;
+  onResize?: () => void;
+  onTiltChanged?: () => void;
+  onZoomChanged?: () => void;
+  onLoad?: (map: google.maps.Map) => void;
+}
+
+export class GoogleMap extends React.PureComponent<GoogleMapProps, GoogleMapState> {
```

Comment:
I think they will still get useful prop info in their ide even if they don't use typescript (at least in vscode for sure). Regarding PropTypes. I'm not sure how it works right now, but don't they get stripped at build time anyway?

Reply:
Yea, propTypes are DEV only and removed in prod builds. It's a bit different as it checks in runtime instead of compile time. Original react-google-maps does not care about propTypes either, so let's forget about it :)